_posts/2020-05-04-strangler-pattern.md

```diff
@@ -0,0 +1,107 @@
+---
+layout: post
+title: Strangler Pattern
+date: 2020-05-04
+tags: architecture patterns
+author: Tim Myerscough
+image: img/strangler-pattern/Strangler_fig_boulder_katandra.jpg
+---
+
+<center><a href="https://commons.wikimedia.org/wiki/File:Strangler_fig_boulder_katandra.jpg"> <img alt="strangler fig" src="/img/strangler-pattern/Strangler_fig_boulder_katandra.jpg" /></a></center>
+<br/>
+
+Technology is constantly evolving, so thinking in terms of the end goal is unrealistic.  Many companies struggle to keep up with the advances in technology.  If companies lag behind it increases commercial pressure from nimble new entrants, or competitors that can move faster.
+
+Evolutionary architecture, doesn't aim for an end state.  One tool for evolutionary architecture is the [Strangler Pattern](https://martinfowler.com/bliki/StranglerFigApplication.html).
+
+# The Context
+When considering migration to the cloud, [Infrastructure Modernisation](https://www.mechanicalrock.io/our-expertise/infrastructure-modernisation), or [Serverless technology](https://www.mechanicalrock.io/our-expertise/cloud-native-solutions) how can we move faster whilst dealing with legacy systems?
+
+You can't stop everything, down tools and re-write from scratch.  What's more, if you tried you would likely be caught out by the [Second System Effect](https://wiki.c2.com/?SecondSystemEffect).  "We can solve the issues of the past and re-write the system that took us 6+ years, in 12 months."  The reality?  18 months down the line: you are still building on the original system; the new system supports ~20% of the functionality of the original and technical debt accrued in the push deliver has turned your beautiful clean new architecture into a big ball of mud.
+
+Instead, you need a mechanism to refactor and replace parts of a system, whilst maintaining production availability.
+
+# The solution
+
+The strangler pattern provides an approach to controlled re-architecture.  
+
+It is not intended persistent indefinitely.  Instead it provides a pathway to manage internal change whilst minimising the impact to external dependencies.  Once re-architecture has been completed, a key step is the removal of the strangler from your system to simplify and move forward.
```

Comment:
"not intended to persist"

Reply:
Vague. You need to spell out the rewrite fallacy better - and the link doesn't do you any favours. Possibly give examples, find examples in the literature of abandoned system rewrites. There must be many.

Reply:
Also if you your going to start a sentence with a number, spell it out "Eighteen" not "18"

Reply:
"whilst minimising the impact to external dependencies" is a bit obscure. External systems? External concerns? Preserving legacy interfaces?